zendesk/resource_zendesk_ticket_field.go

```diff
@@ -229,6 +242,31 @@ func readTicketField(d identifiableGetterSetter, zd client.TicketFieldAPI) error
 		"agent_description":     field.AgentDescription,
 	}
 
+	// set system field options
+	systemFieldOptions := make([]map[string]interface{}, len(field.SystemFieldOptions))
+	for _, v := range field.SystemFieldOptions {
+		m := map[string]interface{}{
+			"name":  v.Name,
+			"value": v.Value,
+		}
+		systemFieldOptions = append(systemFieldOptions, m)
+	}
+
+	fields["system_field_options"] = systemFieldOptions
```

Comment:
According to Zendesk API document, `system_field_options` is 

> Presented for a ticket field of type "tickettype", "priority" or "status"

Why don't we omit this block when the field is not relevant to them, like

```go
switch field.Type {
case "tickettype", "priority" or "status":
  // above code
default: // nop
}
```

Reply:
The same applies to `custom_field_options` and `regexp_for_validation`.

- `custom_field_options` : only for `"tagger"` type
- `regexp_for_validation` : only for `"regexp"` type

Reply:
i guess my question only question is" why add the logic." Presumably this will just end up being an empty list for any type that this is not valid for.

Reply:
i guess my main concern with this is that i feel like this method is trying to take whatever comes back from the zendesk api and serialize it into a tf resource. Seems like if the api said that there are custom fields this package should respect that since the api is the ultimate source of truth.

Reply:
I understand. You meant this package should have flexibility that independent from API specifications, didn't you?
I agree with you.